azurerm/internal/services/privatedns/resource_arm_private_dns_txt_record.go

```diff
@@ -0,0 +1,237 @@
+		return fmt.Errorf("Error creating/updating Private DNS TXT Record %q (Zone %q / Resource Group %q): %s", name, zoneName, resGroup, err)
+	}
+
+	resp, err := client.Get(ctx, resGroup, zoneName, privatedns.TXT, name)
+	if err != nil {
+		return fmt.Errorf("Error retrieving Private DNS TXT Record %q (Zone %q / Resource Group %q): %s", name, zoneName, resGroup, err)
+	}
+
+	if resp.ID == nil {
+		return fmt.Errorf("Cannot read Private DNS TXT Record %s (resource group %s) ID", name, resGroup)
+	}
+
+	d.SetId(*resp.ID)
+
+	return resourceArmPrivateDnsTxtRecordRead(d, meta)
+}
+
+func resourceArmPrivateDnsTxtRecordRead(d *schema.ResourceData, meta interface{}) error {
+	dnsClient := meta.(*clients.Client).PrivateDns.RecordSetsClient
+	ctx, cancel := timeouts.ForRead(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resGroup := id.ResourceGroup
+	name := id.Path["TXT"]
+	zoneName := id.Path["privateDnsZones"]
+
+	resp, err := dnsClient.Get(ctx, resGroup, zoneName, privatedns.TXT, name)
+	if err != nil {
+		if utils.ResponseWasNotFound(resp.Response) {
+			d.SetId("")
+			return nil
+		}
+		return fmt.Errorf("Error reading Private DNS TXT record %s: %+v", name, err)
+	}
+
+	d.Set("name", name)
+	d.Set("resource_group_name", resGroup)
+	d.Set("zone_name", zoneName)
+	d.Set("ttl", resp.TTL)
+	d.Set("fqdn", resp.Fqdn)
+
+	if err := d.Set("record", flattenAzureRmPrivateDnsTxtRecords(resp.TxtRecords)); err != nil {
+		return err
+	}
+
+	return tags.FlattenAndSet(d, resp.Metadata)
+}
+
+func resourceArmPrivateDnsTxtRecordDelete(d *schema.ResourceData, meta interface{}) error {
+	dnsClient := meta.(*clients.Client).PrivateDns.RecordSetsClient
+	ctx, cancel := timeouts.ForDelete(meta.(*clients.Client).StopContext, d)
+	defer cancel()
+
+	id, err := azure.ParseAzureResourceID(d.Id())
+	if err != nil {
+		return err
+	}
+
+	resGroup := id.ResourceGroup
+	name := id.Path["TXT"]
+	zoneName := id.Path["privateDnsZones"]
+
+	_, err = dnsClient.Delete(ctx, resGroup, zoneName, privatedns.TXT, name, "")
+	if err != nil {
+		return fmt.Errorf("Error deleting Private DNS TXT Record %s: %+v", name, err)
+	}
+
+	return nil
+}
+
+func flattenAzureRmPrivateDnsTxtRecords(records *[]privatedns.TxtRecord) []map[string]interface{} {
+	results := make([]map[string]interface{}, 0)
+
+	if records != nil {
+		for _, record := range *records {
+			txtRecord := make(map[string]interface{})
+
+			if v := record.Value; v != nil {
+				value := strings.Join(*v, "")
+				txtRecord["value"] = value
+			}
+
+			results = append(results, txtRecord)
+		}
+	}
+
+	return results
+}
+
+func expandAzureRmPrivateDnsTxtRecords(d *schema.ResourceData) *[]privatedns.TxtRecord {
+	recordStrings := d.Get("record").(*schema.Set).List()
+	records := make([]privatedns.TxtRecord, len(recordStrings))
+
+	segmentLen := 254
+	for i, v := range recordStrings {
+		record := v.(map[string]interface{})
```

Comment:
I'm always hesitant about making this type of assertion as I've seen cases where this is nil and we panic. Let's add a continue if this is nil and then we can keep the rest of the logic the same.

Reply:
Do you mean the record values may come back as nil? e.g. mitigate like:

```go
	segmentLen := 254
	for i, v := range recordStrings {
		if v == nil {
			continue
		}

		record := v.(map[string]interface{})
```

Reply:
Exactly. Looks great!